### PR TITLE
Add annotations for setting cpu/memory limits on sidecar

### LINKED ISSUE #16126: Setting istio sidecar proxy resource request/limit
Hello :)

**Describe the feature request**

Is there an option to set istio proxy sidecar request/limits per application (Pod)? For some of our applications we would like to use Guaranteed QoS (https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/) that require setting `limit == request` for all Pod containers. Without such functionality we're loosing this feature.

**Describe alternatives you've considered**

We've considered using global config value (https://istio.io/docs/reference/config/installation-options/#global-options) but if I understood correctly it could be used only during Istio installation, not applied for each Pod separately what don't match described case.
Some of our applications have bigger traffic so need higher requested resource values, other lower. As istio-proxy resources are somehow proportional to application container resources (e.g. one application is handling much bigger traffic than another -> it will require more resources also on istio-proxy sidecar layer) I would like to have an option to align resources assigned to isito-proxy per Pod.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[X] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### PR DESCRIPTION
**Pull request by rolandkool**
When a limitrange is active, not setting the limits will result
in an error. This patch will allow setting limits for the sidecar.

Fixes #16126

Change-Id: I031d510812a867c2790eaa9af2f51145b4e5f006

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
lgtm. Should better add a test for it.

**Review by hzxuzhonghu — APPROVED** [review]
Thanks

**Comment by rolandkool** [comment]
@hzxuzhonghu I've pushed a commit to update the test. Is this sufficient? If not, could you provide some pointers where to add/fix tests?

**Comment by rolandkool** [comment]
Do I also need to prepare a PR for the documentation?

**Review by howardjohn — APPROVED** [review]
Thank you!

**Comment by rolandkool** [comment]
@howardjohn thanks for approval. Any chance this could make it into 1.5.2?

**Comment by stewartbutler** [comment]
@howardjohn Looks like the cherrypick failed, I created a manual cherrypick here but the diff is pretty large https://github.com/istio/istio/pull/22444

I think the large diff is due to mixer being removed in master?

File: pkg/kube/inject/testdata/webhook/resource_annotations.yaml [hunk 1]
```diff
@@ -11,7 +11,9 @@ spec:
     metadata:
       annotations:
         sidecar.istio.io/proxyCPU: "100m"
+        sidecar.istio.io/proxyCPULimit: "1000m"
```
Comment (ruban-rengaraju) [inline pkg/kube/inject/testdata/webhook/resource_annotations.yaml:14]: Very useful!

**Pull request merged**